src/rtp/rtp.c

```diff
@@ -26,19 +27,19 @@
 struct rtp_sock {
 	/** Encode data */
 	struct {
-		uint16_t seq;   /**< Sequence number       */
-		uint32_t ssrc;  /**< Synchronizing source  */
+		uint16_t seq;                /**< Sequence number            */
+		uint32_t ssrc;               /**< Synchronizing source       */
 	} enc;
-	int proto;              /**< Transport Protocol    */
-	void *sock_rtp;         /**< RTP Socket            */
-	void *sock_rtcp;        /**< RTCP Socket           */
-	struct sa local;        /**< Local RTP Address     */
-	struct sa rtcp_peer;    /**< RTCP address of Peer  */
-	rtp_recv_h *recvh;      /**< RTP Receive handler   */
-	rtcp_recv_h *rtcph;     /**< RTCP Receive handler  */
-	void *arg;              /**< Handler argument      */
-	struct rtcp_sess *rtcp; /**< RTCP Session          */
-	bool rtcp_mux;          /**< RTP/RTCP multiplexing */
+	int proto;                           /**< Transport Protocol         */
+	void *sock_rtp;                      /**< RTP Socket                 */
+	void *sock_rtcp;                     /**< RTCP Socket                */
+	struct sa local;                     /**< Local RTP Address          */
+	struct sa rtcp_peer;                 /**< RTCP address of Peer       */
+	rtp_recv_h *recvh;                   /**< RTP Receive handler        */
+	rtcp_recv_h *rtcph;                  /**< RTCP Receive handler       */
+	void *arg;                           /**< Handler argument           */
+	struct rtcp_sess *rtcp;              /**< RTCP Session               */
+	RE_ATOMIC bool rtcp_mux;             /**< RTP/RTCP multiplexing      */
```

Comment:
is this the only field shared between threads when using multi-threading ?

Is it possible to put the lock in the application code instead ?

if more fields are shared, it is probably better to use a mutex lock.

Reply:
> is this the only field shared between threads when using multi-threading ?

Yes, the setter `rtcp_enable_mux()` is called from main thread, while the RTP RX thread has to check this bool during processing of incoming RTP packets. The clang sanitizer reported this.

> Is it possible to put the lock in the application code instead ?

I don't think so, because in baresip the `stream->rtp` is used at different places and also passed to helpers and so on.

> if more fields are shared, it is probably better to use a mutex lock.

I try to keep the locking minimal. Here is the list of potential concurrency problems I collected by analyzing the code.c: 
- stream.c: https://github.com/baresip/baresip/discussions/2445#discussioncomment-4903505
- audio.c: https://github.com/baresip/baresip/discussions/2445#discussioncomment-4903704